spec/lib/api/v3/work_packages/work_package_representer_spec.rb

```diff
@@ -715,6 +717,60 @@
         end
       end
 
+      describe 'fileLinks' do
+        let(:storages_feature_enabled) { true }
+
+        before do
+          allow(OpenProject::FeatureDecisions).to receive(:storages_module_active?).and_return(storages_feature_enabled)
+        end
+
+        it_behaves_like 'has an untitled link' do
+          let(:link) { 'fileLinks' }
+          let(:href) { api_v3_paths.file_links(work_package.id) }
+        end
+
+        it_behaves_like 'has an untitled action link' do
+          let(:permission) { :manage_file_links }
+          let(:link) { 'addFileLink' }
+          let(:href) { api_v3_paths.file_links(work_package.id) }
+          let(:method) { 'post' }
+        end
+
+        context 'if user has no permission to view file links' do
+          let(:permissions) { all_permissions - %i[view_file_links] }
+
+          it_behaves_like 'has no link' do
+            let(:link) { 'fileLinks' }
+          end
+        end
+
+        context 'if file storages feature is disabled' do
+          let(:storages_feature_enabled) { false }
+
+          it_behaves_like 'has no link' do
+            let(:link) { 'fileLinks' }
+          end
+
+          it_behaves_like 'has no link' do
+            let(:link) { 'addFileLink' }
+          end
+        end
+
+        context 'if project has storages module disabled' do
```

Comment:
Do we really need to test if feature flags work here? Feature flags usually come and go.

Reply:
valid question. One, I cannot answer on my own. Will raise a general question in devs channel

Reply:
feature flags should not be tested at that level here, as everything should be hidden behind the permission check itself